### PR TITLE
gh-99180: Remove traceback anchors in return and assign statements that cover all the displayed range

### PR DESCRIPTION
**Pull request by pablogsal**
* Issue: gh-99180

### CODE REVIEW

File: Lib/test/test_traceback.py [hunk 1]
```diff
@@ -1487,6 +1486,186 @@ def f():
             '    raise MemoryError()']
         self.assertEqual(actual, expected)
 
+    def test_anchors_for_simple_return_statements_are_eluded(self):
```
Comment (AlexWaygood) [inline Lib/test/test_traceback.py]:
```suggestion
    def test_anchors_for_simple_return_statements_are_elided(self):
```

File: Lib/test/test_traceback.py [hunk 2]
```diff
@@ -1487,6 +1486,186 @@ def f():
...
+            "             ^^",
+            f"  File \"{__file__}\", line {g.__code__.co_firstlineno + 1}, in g",
+            "    1/0",
+            "    ~^~"
+        ]
+        self.assertEqual(result_lines, expected)
+
+
+
+    def test_anchors_for_simple_assign_statements_are_eluded(self):
```
Comment (AlexWaygood) [inline Lib/test/test_traceback.py]:
```suggestion
    def test_anchors_for_simple_assign_statements_are_elided(self):
```

**Review by AlexWaygood — COMMENTED** [review]
Spotted some minor typos:

**Comment by pablogsal** [comment]
Hey @lysnikolaou, can you check if this is more or less what you had in mind? (I will rebase soon)

**Review by lysnikolaou — APPROVED** [review]
This looks a lot like what I had in mind and seeing the implementation I think it's a step towards the right direction!

**Comment by lysnikolaou** [comment]
@pablogsal Let's merge this?

File: Lib/traceback.py [hunk 3]
```diff
@@ -727,6 +724,37 @@ def output_line(lineno):
 
         return ''.join(row)
 
+    def should_show_carets(self, start_offset, end_offset, all_lines, anchors):
```
Comment (AlexWaygood) [inline Lib/traceback.py:727]: Is this method meant to be a public-API method on `StackSummary` objects, or is it meant to be private API? If the former, we should probably document it at https://docs.python.org/3.13/library/traceback.html#stacksummary-objects. If the latter, maybe it should have a name prefixed with an underscore?

Comment (pablogsal) [reply Lib/traceback.py:727]: It should be private so I am fine prefixing it with an underscore

Comment (AlexWaygood) [reply Lib/traceback.py:727]: https://github.com/python/cpython/pull/119554

**Pull request merged**